common/data_refinery_common/models/models.py

```diff
@@ -789,14 +789,24 @@ def needs_processing(self, own_processor_id=None) -> bool:
             return False
 
         if sample.source_database == "SRA":
-            for computed_file in sample.computed_files.prefetch_related('result__organism_index').all():
-                    if computed_file.s3_bucket and computed_file.s3_key \
-                       and computed_file.result.organism_index != None \
-                       and computed_file.result.organism_index.salmon_version == CURRENT_SALMON_VERSION:
-                        # If the file wasn't computed with the latest
-                        # version of salmon, then it should be rerun
-                        # with the latest version of salmon.
-                        return False
+            computed_file = sample.get_most_recent_smashable_result_file()
+
+            # If there's no smashable file then we should check the quant.sf file.
+            if not computed_file:
+                computed_file = sample.get_most_recent_quant_sf_file()
+
+            # If there's neither a quant.sf file nor a smashable file
+            # then we definitely need to process it.
+            if not computed_file:
+                return True
+
+            if computed_file.s3_bucket and computed_file.s3_key \
```

Comment:
I feel tempted to suggest to add a new method to `ComputedFile` to check for this. Like, `is_uploaded_to_s3`. We can add it later if we think it's helpful